core/src/main/java/io/grpc/transport/ServerStreamListener.java

```diff
@@ -31,10 +31,17 @@
 
 package io.grpc.transport;
 
+import io.grpc.MethodType;
 import io.grpc.Status;
 
 /** An observer of server-side stream events. */
 public interface ServerStreamListener extends StreamListener {
+
+  /**
+   * The {@link MethodType} for the call.
+   */
+  MethodType getMethodType();
```

Comment:
This is how the server stream knows the difference between unary & non-unary calls for flush optimization. It has to be on the listener as this is only known by the application layer and is resolved after server stream creation.